litestar/logging/config.py

```diff
@@ -49,7 +49,17 @@
 }
 
 if sys.version_info >= (3, 12, 0):
-    default_handlers["queue_listener"]["handlers"] = ["console"]
+    default_handlers["queue_listener"].update(
+        {
+            "class": "logging.handlers.QueueHandler",
+            "queue": Queue(-1),
```

Comment:
Instantiating a `queue.Queue` object in the handler config might be the source of the "test suite hanging" problem.

Reply:
Perhaps - maybe try marking all of those logging tests with `pytest.mark.xdist("something")` so they all run in the same process and see if that helps?

Reply:
In fact, I guess `litestar/logging/config.py` and so `default_handlers` might be loaded in different tests and context. Instanciating a `Queue` object directly in the config seems to be a bad idea in most cases.

Instead, as mentioned in the Python doc, I can also use a factory. From what I've seen locally, it fixes the problem. I'm going to push the change, but not sure how I can run the test suite multiple times on Github Actions to check if everything is fine... I can just amend the commit without any change to refresh the commit id. But maybe you'd have a better idea?

Reply:
> I can just amend the commit without any change to refresh the commit id. But maybe you'd have a better idea?

You can also just push an empty commit if you want to trigger a rerun